### ISSUE TITLE
XmlSeeAlso in combination with episodal compilation

### ISSUE THREAD

**Issue by sjaakd** [highsource/jaxb2-basics#103]
I'm compiling opengis schemas like sampling feature and observation and measurement. I've created a set of util libraries on that level uses the generated jar files. So far so good.

However, now observation and measurement / sampling feature are actually used within a new library, lets say library X. So I have classes in library X that extend OM_Observation and SF_SamplingFeature.

So far so good.

However, my CXF stack in JBOSS can't parse documents that expect an OM_Observation or SF_SamplingFeature. I suspect because if I don't use episodal compilation, the `XmlSeeAlso` annotation is not generated. **Note**: I need to investigate this further to be sure.

Now: the sun-jaxb.episode is in essence a binding which dictates that the classes from the sampling-feature-library and the om-observation-library are used. I can't change these libraries to include XmlSeeAlso tags of for instance libary X (that would also be a strange dependency), and the binding in de sun-jaxb.episode does not allow regeneration of the OM_Obeservation of SF_SampingFeature within library X. Feels like a catch22. 

Any idea's how to solve this conundrum?

**Comment by highsource** [comment]
Hi @sjaakd.

First of all, are you aware of https://github.com/highsource/ogc-schemas?
I would suggest that we start by adding a reproducing test case there. OGC schemas are quite hard to compile and there are a lot of tricks one need to employ.

I would rather not rely on `@XmlSeeAlso` but prefer to generate a `@XmlElementRef` or `@XmlAnyElement` properties. This allows JAXB to parse elements defined in the context even if they are not specified in `@XmlSeeAlso`.

But, let's start with a test case in `ogc-schemas` and see where we get from there.

**Comment by highsource** [comment]
Closing the issue here, please open a new issue in `ogc-schemas` once you have a test case.

**Comment by sjaakd** [comment]
Hi @highsource .  I am aware of the libraries. We had contact on some issues with episodes some years back. 

The problem is that we nowadays use an excerpt of OGC XSD's called profiles. The reason: a lot of our clients (also non java) had problems with the OGC XSD's. Also, there's some legacy (5 years of work). That's why we have our own libraries, but quite similar to yours (thanks to this really nice jaxb-plugin). 

In the mean time I disabled episodal compilation for 2 libraries (OM + SF). There's a limit number of classes generated with the desired `@XmlSeeAlso`. What is funny is that there's an `@XmlSeeAlso` on the wsdl interface class. One of the first things I did is add our object factories there (something that normally works when adding to a JAXBContext)  o I could stick to episodal compilation. Unfortunately that didn't seem to work. It could be some CXF / JBOSS quirk. 

I'll for sure checkout how you compile your libraries, to see if I can learn from that. I think that `XmlAnyElement` properties (lax) are generated, but not on the place where I need them.
